### DISCUSSION TITLE
Tensorboard implementation with Heterogenous Graph

### DISCUSSION THREAD

**Discussion by shahfasal** [pyg-team/pytorch_geometric#4744] [Q&A]
Hi Team,
I am using [code example ](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/hetero/hetero_link_pred.py) to do link prediction.    
I have implemented tensorboard functionality to write test and train loss. Now I was trying to add Tensorboard functionality on  the model using tensorboard.add_graph     

    with torch.no_grad():
        model.encoder(train_data.x_dict, train_data.edge_index_dict)  
        writer.add_graph(model, [data.x_dict, data.edge_index_dict])
       
this throws the error

`RuntimeError: Cannot create dict for key type '(str, str, str)', only int, float, complex, Tensor, device and string keys are supported`

tensorboard expects tensor while as encoder is taking dictionary input... How can we resolve this to add our model to tensorboard?
Thanks
cc @rusty1s

**Comment by rusty1s** [comment]
I see. This is indeed a problem. You may be able to fix this by converting the edge type keys into a single string:
```python
edge_index_dict = ['__'.join(edge_type) for edge_type, edge_index in train_data.edge_index_dict.items()]
model.encoder(train_data.x_dict, edge_index_dict)
```
and re-map to a tuple once again inside the model.
